graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/kotlin/KotlinDataTypeGenerator.kt

```diff
@@ -63,6 +63,16 @@ class KotlinInputTypeGenerator(private val config: CodeGenConfig, private val do
                             is StringValue -> CodeBlock.of("%S", defVal.value)
                             is FloatValue -> CodeBlock.of("%L", defVal.value)
                             is EnumValue -> CodeBlock.of("%M", MemberName(type as ClassName, defVal.name))
+                            is ArrayValue -> if(defVal.values.isEmpty()) CodeBlock.of("emptyList()") else CodeBlock.of("listOf(%L)", defVal.values.map { v ->
```

Comment:
IIRC `collection.map { body }.joinToString()` could be rewritten as `collection.joinToString { body }`.

Reply:
Would need to do a toString() on each codeblock in that case.

Reply:
I'm using (in my patch for another comment)
```kotlin
CodeBlock.of("listOf(%L)", value.values.joinToString { v -> generateCode(v, type).toString() })
```
which saves an intermediary list. The call to `toString()` is explicit in this approach, but would happen anyhow (`Iterable<T>.joinToString` uses `Iterable<T>.joinTo` which eventually calls [`Appendable.appendElement`](https://github.com/JetBrains/kotlin/blob/29b23e79f32791e456a5b4a453277f0f0b3e984d/libraries/stdlib/src/kotlin/text/Appendable.kt#L83) which calls `toString()` by default :) )